auth/TESLA/test/TestSenderNotary.m

```diff
@@ -0,0 +1,157 @@
+classdef TestSenderNotary < matlab.unittest.TestCase
+
+    methods (Test)
+
+        function test_key_chain_construction(testCase)
+            sn = Sender_Notary();
+            for i = 1:sn.key_chain_length - 1
+                hash = HashingWrappers.sha_256(sn.key_chain{i});
+                testCase.assertEqual(hash, ...
+                                     sn.key_chain{i + 1});
+            end
+        end
+
+        function test_M50_creation(testCase)
+            sn = Sender_Notary();
+            count = 1;
+            time = 1;
+            for i = 1:1000
+                message = uint8(strcat(['message ', num2str(i)]));
+
+                test = sn.load_new_message(message, time);
+
+                if isa(test, 'MT50')
+                    testCase.assertEqual(test.hash_point, sn.key_chain{end - count + 1});
+                    count = count + 1;
+                    if count >= sn.key_chain_length
+                        count = 1;
+                    end
+                end
+                time = time + 1;
+            end
+
+        end
+
+        function test_next_keychain(testCase)
+            sn = Sender_Notary();
+            path1 = Sender_Notary.copy_keychain(sn.key_chain);
+            path2 = sn.next_key_chain;
+            time = 1;
+            count = 1;
+
+            for i = 1:700
+                message = uint8(strcat(['message ', num2str(i)]));
+
+                test = sn.load_new_message(message, time);
+
+                if isa(test, 'MT50')
+                    testCase.assertEqual(test.hash_point, sn.key_chain{end - count + 1});
+                    count = count + 1;
+                    if count >= sn.key_chain_length
+                        count = 1;
+                    end
+                end
+                time = time + 1;
+            end
+            testCase.assertNotEqual(path1, sn.key_chain);
+            testCase.assertEqual(path2, sn.key_chain);
+
+        end
+
+        function test_sender_and_reciever(testCase)
+            sn = Sender_Notary();
+            pass = sn.get_hash_path_end;
+            pass2 = sn.get_next_hash_path_end;
+            rn = Receiver(pass, pass2);
+
+            time = 1;
+
+            for i = 1:100
+                message = uint8(strcat(['message ', num2str(i)]));
+
+                m = sn.load_new_message(message, time);
+                rn.add_message(message, time);
+                if isa(m, 'MT50')
```

Comment:
I don't see any difference in behavior if the message is type `MT50`...why is this if branch necessary?

Reply:
m will be an mt 50 after every 5th message and null otherwise so you load in the mt 50 and increment the time interval for that step

Reply:
Ah, I read `m` and `message` as the same variable.